File: php/muse/algorithms/meta/SearchTest.php
<?php
    require_once "muse/algorithms/meta/Search.php";
    require_once "muse/util/SequenceBuilder.php";
    require_once "muse/util/TestRunner.php";

    use muse\algorithms\meta\Search;
    use muse\util\SequenceBuilder;
    use muse\util\TestRunner;

    function testBinarySearch(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packIncreasing($arr);

        if (Search\binarySearch($arr, -1) != $size) {
            return false;
        }

        if (Search\binarySearch($arr, 2147483647) != $size) {
            return false;
        }

        foreach ($arr as $i => $v) {
            if (Search\binarySearch($arr, $v) != $i) {
                return false;
            }
        }

        return true;
    }

    function testLinearSearch(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packIncreasing($arr);

        if (Search\linearSearch($arr, -1) != $size) {
            return false;
        }

        if (Search\linearSearch($arr, 2147483647) != $size) {
            return false;
        }

        foreach ($arr as $i => $v) {
            if (Search\linearSearch($arr, $v) != $i) {
                return false;
            }
        }

        return true;
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\pick(testBinarySearch);

        TestRunner\pick(testLinearSearch);
    }
?>

File: java/muse/algorithms/SearchTest.java
import muse.algorithms.Search;
import muse.util.SequenceBuilder;
import muse.util.TestRunner;

public class SearchTest {

  public static boolean testBinarySearch() {
    int size = 32768;

    Integer[] arr = new Integer[size];
    SequenceBuilder.packIncreasing(arr);

    if (Search.binarySearch(arr, -1) != size) {
      return false;
    }

    if (Search.binarySearch(arr, 2_147_483_647) != size) {
      return false;
    }

    for (int i = 0; i < size; i++) {
      if (Search.binarySearch(arr, arr[i]) != i) {
        return false;
      }
    }

    return true;
  }

  public static boolean testLinearSearch() {
    int size = 32768;

    Integer[] arr = new Integer[size];
    SequenceBuilder.packIncreasing(arr);

    if (Search.linearSearch(arr, -1) != size) {
      return false;
    }

    if (Search.linearSearch(arr, 2_147_483_647) != size) {
      return false;
    }

    for (int i = 0; i < size; i++) {
      if (Search.linearSearch(arr, arr[i]) != i) {
        return false;
      }
    }

    return true;
  }

  public static void main(String[] args) {
    TestRunner.pick(SearchTest::testBinarySearch);

    TestRunner.pick(SearchTest::testLinearSearch);
  }
}
